src/pages/home/report/ReportActionsView.js

```diff
@@ -132,7 +133,9 @@ class ReportActionsView extends React.Component {
 
         // Only mark as read if the report is open
         if (!this.props.isDrawerOpen) {
-            updateLastReadActionID(this.props.reportID);
+            if (isReportDataUptoDate()) {
```

Comment:
Just check the value from Onyx.key ONYXKEYS.INITIAL_REPORT_DATA_LOADED and it should work.